From 00ebea95ae13ba4402dd1382f74b8b6af6cebd6f Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Tue, 21 Jul 2015 03:25:59 +0800
Subject: [PATCH 1049/1221] MLK-11264 ARM: imx: correct the MMDC bypass
 reorder on imx6qp

In low busfreq mode, set the MMDC0_MAARCR bit25~26 to
2b'01(Disable MMDC reorder), set to 2b'10 (Bypass MMDC ADOPT)
in high busfreq mode.

Signed-off-by: Bai Ping <b51503@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/mach-imx/ddr3_freq_imx6.S |   19 ++++++++-----------
 1 files changed, 8 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-imx/ddr3_freq_imx6.S b/arch/arm/mach-imx/ddr3_freq_imx6.S
index febc15c..dd689e4 100644
--- a/arch/arm/mach-imx/ddr3_freq_imx6.S
+++ b/arch/arm/mach-imx/ddr3_freq_imx6.S
@@ -523,6 +523,14 @@ poll_conreq_set_1:
 	cmp	r0, #(0x4 << 12)
 	bne	poll_conreq_set_1
 
+	/*
+	 * if requested frequency is great than
+	 * 300MHz, skip setting bypass adopt mode.
+	 */
+	ldr	r1, =300000000
+	cmp	r4, r1
+	bge	1f
+
 	is_mx6qp
 	bne	1f
 	/* Switch to adopt mode, set MMDC0_MAARCR bit25~26 to 2b'01 */
@@ -728,17 +736,6 @@ continue_dll_off_3:
 	bic	r0, r0, #(0x1 << 8)
 	str	r0, [r5, #MMDC0_MADPCR0]
 
-	is_mx6qp
-	bne	2f
-	/*
-	 * Switch back to adopt_bp mode, set MMDC0_MAARCR
-	 * bit25~26 to 2b'10.
-	 */
-	ldr	r0, [r5, #MMDC0_MAARCR]
-	bic	r0, r0, #(0x3 << 25)
-	orr	r0, r0, #(0x2 << 25)
-	str	r0, [r5, #MMDC0_MAARCR]
-2:
 	mov	r0, #0x0
 	str	r0, [r5, #MMDC0_MDSCR]
 poll_conreq_clear_1:
-- 
1.7.5.4

